pos/app/(main)/report/components/timePicker.tsx

```
@@ -0,0 +1,108 @@
+export function TimePicker({ value, onChange }: TimePickerProps) {
+  const isValidTime = (time: string) => /^(\d{2}):(\d{2})$/.test(time);
+  const parsedValue = isValidTime(value) ? value : "00:00";
+
+  const hours = Number.parseInt(parsedValue.split(":")[0]);
+  const minutes = Number.parseInt(parsedValue.split(":")[1]);
+  const period = hours >= 12 ? "PM" : "AM";
+  const hour12 = hours === 0 ? 12 : hours > 12 ? hours - 12 : hours;
+
+  const handleHourChange = (newHour: string) => {
+    const hourNumber = Number.parseInt(newHour);
+    if (!isNaN(hourNumber)) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)